### PR TITLE
adding prototype for tensoboard projector to visualize the embeddings with UMAP,TSNE, PCA/

### PR DESCRIPTION
**Pull request by edyoshikun**

### CODE REVIEW

File: viscy/representation/engine.py [hunk 1]
```diff
@@ -168,37 +252,122 @@ def training_step(self, batch: TripletSample, batch_idx: int) -> Tensor:
...
         pos_img = batch["positive"]
         anchor_projection = self(anchor)
         positive_projection = self(pos_img)
         negative_projection = None
+
         if isinstance(self.loss_function, NTXentLoss):
-            indices = torch.arange(
-                0, anchor_projection.size(0), device=anchor_projection.device
-            )
+            batch_size = anchor.size(0)
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply viscy/representation/engine.py:323]: (I didn't know that you can pass an argument to `size()`!)
I've been using `Tensor.shape[0]` so should be equivalent.

File: viscy/representation/engine.py [hunk 2]
```diff
@@ -168,37 +252,122 @@ def training_step(self, batch: TripletSample, batch_idx: int) -> Tensor:
...
+            indices = torch.arange(0, batch_size, device=anchor_projection.device)
             labels = torch.cat((indices, indices))
             # Note: we assume the two augmented views are the anchor and positive samples
             embeddings = torch.cat((anchor_projection, positive_projection))
             loss = self.loss_function(embeddings, labels)
             self._log_step_samples(batch_idx, (anchor, pos_img), "val")
+
+            # Store embeddings for visualization
+            if self.current_epoch % self.embedding_log_interval == 0 and batch_idx == 0:
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply viscy/representation/engine.py:332]: Would that always be the first time point(s) then?

[the remark this answers is not in the document]
Comment (ziw-liu) [reply viscy/representation/engine.py:332]: Note that this will not be a representative (i.e. uniformly sampled) subset of the validation set. For example, in an infection time-lapse, all logged examples will be uninfected.
